### PR TITLE
docs: refactor react-navigation example with React Nav team feedback

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Refactor React Navigation example based on [feedback](https://github.com/callstack/react-native-testing-library/pull/1237#discussion_r1033973998) from @satya164 from React Navigation team.

### CODE REVIEW

File: examples/react-navigation/README.md [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+
+## Non-recommended tests
+
+There also exists another popular type of screen level tests, where users mock React Navigation objects like `navigation`, `route` and/or hooks like  `useNavigation`, etc. This type is not recommended by the React Navigation team as it leads to fragile tests that can lead to false positive and false negative result. While the recommended approach is arguable harder it should lead to more robust tests helping to properly identify potential issues in test. 
 
 > Note that this example applies `includeHiddenElements: false` by default, so all queries will ignore elements on the hidden screens, e.g. inactive tabs or screens present in stack navigators. This option is enabled in `jest-setup.js` file, using `defaultIncludeHiddenElements: false` option to `configure` function.
```
Comment (thymikee) [inline examples/react-navigation/README.md:16]: this ended up in "non-recommended tests" section now. can you move it up?

File: examples/react-navigation/README.md [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+
+## Recommended tests
+
+There are two types of recommeded tests:
+1. Tests operating on navigator level - these use `renderNavigator` helper to render a navigator component used in the app. It is useful when you want to test a scenario that includes multiple screens.
+2. Tests operating on single screen level - these use regular `render` helper but require refactoring screen components into `Screen` and `ScreenContent` components. Where `Screen` receives React Navigation props and/or uses hooks like `useNavigation` while `ScreenContent` does not have a direct relation to React Navigation API but gets props from `Screen` and calls relevant callbacks to trigger navigation.
+
+## Non-recommended tests
+
+There also exists another popular type of screen level tests, where users mock React Navigation objects like `navigation`, `route` and/or hooks like  `useNavigation`, etc. This type is not recommended by the React Navigation team as it leads to fragile tests that can lead to false positive and false negative result. While the recommended approach is arguable harder it should lead to more robust tests helping to properly identify potential issues in test.
```
Comment (thymikee) [inline examples/react-navigation/README.md:14]: Consider rephrasing like this. I'm linking to TL's guiding principles and also not mentioning that our testing setup is "arguably harder", it's counter productive
```suggestion
There also exists another popular type of screen level tests, where users mock React Navigation objects like `navigation`, `route` and/or hooks like  `useNavigation`, etc. We don't recommend this way of testing. **Mocking internal parts of the libraries is effectively testing implementation details, which goes against the Testing Library's [Guiding Principles](https://testing-library.com/docs/guiding-principles/)**.
```

**Review by thymikee — APPROVED** [review]
Needs some small adjustments, other than that looks good!

File: examples/react-navigation/README.md (hunk 1 above)
Comment (mdjastrzebski) [reply examples/react-navigation/README.md]: Moved it up

**Pull request merged**